File: java/src/main/java/deltix/luminary/ItemDef.java
package deltix.luminary;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.List;

public abstract class ItemDef {
    private final String name;
    private final String fullName;
    private final List<String> comments = new ArrayList<>();
    private final List<Decorator> decorators = new ArrayList<>();

    protected ItemDef(@NotNull String name, @NotNull String fullName, @Nullable List<String> comments) {
        this.name = name;
        this.fullName = fullName;
        if (comments != null)
            this.comments.addAll(comments);
    }

    @NotNull
    public String getName() {
        return name;
    }

    @NotNull
    public String getFullName() {
        return fullName;
    }

    @NotNull
    public List<Decorator> getDecorators() {
        return decorators;
    }

    @NotNull
    public List<String> getComments() {
        return comments;
    }
}

File: csharp/main/ItemDef.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary
{
	public abstract class ItemDef : ICommentable, IDecoratable
	{
		protected ItemDef([NotNull] String name, [CanBeNull, ItemNotNull] List<String> comments)
		{
			Name = name;
			if (comments != null)
				Comments.AddRange(comments);
		}

		[NotNull]
		public String Name { get; }

		[NotNull, ItemNotNull]
		public List<Decorator> Decorators { get; } = new List<Decorator>();

		[NotNull, ItemNotNull]
		public List<String> Comments { get; } = new List<String>();

		[NotNull]
		public abstract String FullName { get; }
	}
}
